### PR TITLE
[Azure Search] Regenerate Autocomplete SDK from the swagger spec. 

### PR DESCRIPTION
**Pull request by mhko**
Regenerate Autocomplete SDK from the swagger spec that improves consistency in APIs. 

https://github.com/Azure/azure-rest-api-specs/pull/3268 

@brjohnstmsft, @updixit, @Yahnoosh 

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Customizations/Documents/DocumentsOperationsExtensions.cs [hunk 1]
```diff
@@ -27,9 +27,10 @@ public static class DocumentsOperationsExtensions
         /// </param>
         public static long Count(
             this IDocumentsOperations operations,
-            SearchRequestOptions searchRequestOptions = default(SearchRequestOptions))
+            SearchRequestOptions searchRequestOptions = default(SearchRequestOptions),
+            Dictionary<string, List<string>> customHeaders = null)
```
Comment (brjohnstmsft) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Customizations/Documents/DocumentsOperationsExtensions.cs]: Please make these consistent with the conventions used by the generated methods that we expose in the rest of the SDK. These only take `customHeaders` in the core `WithHttpMessagesAsync` methods. Take a look [here](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Generated/DataSourcesOperationsExtensions.cs) for examples.

**Review by bitsforthought — CHANGES_REQUESTED** [review]
Custom headers are actually only used in the `WithHttpMessagesAsync` methods

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
Approved, will merge when SDK is generated with the merged spec.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Customizations/Documents/DocumentsOperationsExtensions.cs [hunk 2]
```diff
@@ -66,15 +66,12 @@ public static class DocumentsOperationsExtensions
         /// </param>
         public static AutocompleteResult Autocomplete(
             this IDocumentsOperations operations, 
-            AutocompleteMode autocompleteMode,
-            string search,
+            string searchText,
             string suggesterName = null,
```
Comment (brjohnstmsft) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Customizations/Documents/DocumentsOperationsExtensions.cs]: Shouldn't this be required?

**Review by bitsforthought — CHANGES_REQUESTED** [review]

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Customizations/Documents/DocumentsOperationsExtensions.cs (hunk 2 above)
Comment (mhko) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Customizations/Documents/DocumentsOperationsExtensions.cs]: The Autocomplete`Mode` parameter moved to AutocompleteParameters and is optional. It defaults to the oneTermMode by the service if not specified. There's one added test case that tests the default mode.

Comment (mhko) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Customizations/Documents/DocumentsOperationsExtensions.cs]: Good catch. Thanks.

**Review by bitsforthought — APPROVED** [review]

**Review by bitsforthought — APPROVED** [review]

**Pull request merged**